content/guides/getstarted/rust-to-solana.md

```diff
@@ -0,0 +1,273 @@
+third-party crates are supported as well, there are several limitations. Since
+the Solana runtime has resource constraints and must run deterministically, here
+are the differences to be aware of:
+
+### Package Limitations
+
+The following packages are unavailable:
+
+- rand
+- std::fs
+- std::net
+- std::future
+- std::process
+- std::sync
+- std::task
+- std::thread
+- std::time
+
+The following packages have limited functionality:
+
+- std::hash
+- std::os
+
+### Rand Dependencies
+
+Because programs must run deterministically, the `rand` crate is not available.
+Using an additional crate that depends on `rand` will also cause compile errors.
+
+However, if the crate used simply depends on `rand` but does not actually
+generate random numbers, then it is possible to work around this by adding the
+following to the program’s Cargo.toml:
+
+```toml
+[dependencies]
+getrandom = { version = "0.1.14", features = ["dummy"] }
+```
+
+### Macro Changes
+
+Several standard macros have been replaced or have modified behavior to be aware
+of. First, the `println!` has been replaced with the computationally simpler
+`msg!` macro. The `msg!` macro outputs to the program logs, and it can be used
+as follows:
+
+```rust
+msg!("Your message");
+msg!(0_64, 1_64, 2_64);
+msg!("Your variable: {:?}", variable);
+```
+
+The `panic!`, `assert!`, and any internal panics are also output to the program
+logs by default. However, this can be modified with a custom panic handler.
+
+As a better alternative to `panic!`, error handling should be used:
+
+```rust
+#[error_code]
+pub enum MyErrors {
+    CustomError,
+}
+
+#[program]
+pub mod anchor_error_test {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        //panic!("PANIC");
+        return Err(MyErrors::CustomError.into());
+
+        Ok(())
+    }
+}
+```
+
+## Compute Budget
+
+As a Rust developer, efficient computing is nothing new. What may be different
+is that in Solana, each transaction has a fixed
+[compute budget](https://solana.com/developers/guides/advanced/how-to-request-optimal-compute)
+that it must not surpass. When transactions exceed the compute budget, they are
+halted and return an error.
+
+Programs can access the number of remaining compute units via the
+`sol_remaining_compute_units` system call, and can log the remaining number of
+compute units with `sol_log_compute_units`.
+
+## Learning the Development Environment and Frameworks
+
+While the Solana CLI and the `solana_program` crate are all that is needed to
+get started, there are a couple of helpful tools which can accelerate learning.
+
+<div align="center">
+    <img src="../../../public/assets/guides/rust-to-solana/network-diagram.png" alt="Network Diagram" style="max-width: 80%; height: auto;"/>
+</div>
+
+### Solana Playground
+
+The [Solana Playground](https://beta.solpg.io/) is a browser-based IDE that
+allows developers to develop and deploy Solana programs.
+
+<div align="center">
```

Comment:
images like this will not be displayed. they should use markdown based image syntax like this

```md
![brief description of the image](/public/assets/image.png)
```

see: https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#images